app/src/androidTest/java/org/connectbot/PubkeyListActivityTest.java

```diff
@@ -106,8 +107,8 @@ public void generateRSAKey() {
 		}
 	}
 
-	public Activity getDisplayedActivityInstance() {
-		final Activity[] currentActivity = new Activity[1];
+	public AppCompatActivity getDisplayedActivityInstance() {
+		final AppCompatActivity[] currentActivity = new AppCompatActivity[1];
```

Comment:
Line 117 below needs to be fixed because it's a Collection<Activity> which can't be converted. Will this work without changing this test file?

Reply:
I think so but not 100% sure...  I've kind of moved on w/my source base--- I'm building with 3.3 canary now API 28 and a ton of new libraries along with the downloadable fonts and pinch to zoom and such.  After rebasing I have to kind of redo everything to get stuff to build again.  Next up is Jetpack/Androidx along with new chromebook-specific bits.  Androidx is the way forward from now on, especially as far as backwards compatibility goes...  BTW this now works [great](https://developer.android.com/topic/arc/optimizing) btw on chromebooks:

```
        <meta-data android:name="WindowManagerPreference:SuppressWindowControlNavigationButton" android:value="true" />
```

Many of the PR submissions are wonky to fit in as-is to my local repo since the code changes dramatically as you move to the newer libraries and such-- or they become outdated if the older submissions aren't committed too.  For example, #615 the "pinch to change font size" one won't patch over the updated clipmanager PR #610 because it is looking for the old clipboard.  The downloadable fonts PR will be inapplicable once the support library is replaced with androidx, etc.  Test code has the same issues as libraries and build tools get updated.

I know you're very conservative with accepting PRs-- but any chance you could either accept/reject some of the submissions currently in the queue or set up a `develop` or `unstable` branch or something where all these can be semi-recklessly combined into a buildable developer version that does build w/ the most recent toolset, targets chromeOS, etc.?  Put another way-- is there a reason you're hesitant to accept these PRs?  There are 29 currently sitting around.  Some can be chucked out at this point, others are probably still pretty good.  But some are in conflict so can't be applied together and growing stale every day.

FWIW when it comes to anything I submit, please absolutely feel free to modify, change, take pieces of or whatever if for whatever reason if it's not fully up to snuff.  I would love to see CB officially building with 3.3, androidx, backwards support- gradle 4.10, newest libraries (is com.takisoft.fix:preference-v7 needed still w/androidx, etc?) with the new fonts stuff etc!

Cheers, and great project.  I've followed it for years and years.

Reply:
Thought:  [the play store was last updated in February of 2017](https://play.google.com/store/apps/details?id=org.connectbot)- could the play store beta or alpha distribution channels be used for a nightly/weekly/monthly build maybe w/all the crazy latest stuff?

Reply:
I hooked up the play store alpha channel to Jenkins last week. It should be uploading all builds that pass the tests.

By the way, do you have an issue with the CLA? That's another check that isn't complete.

Reply:
If you sign the CLA, I can fix up these problems from a clean branch.